lib/from_openstudio/geometry/room.rb

```diff
@@ -64,12 +64,166 @@ def self.properties_from_space(space)
+              openstudio_sch_type_lim.setName('Temperature')
+              openstudio_sch_type_lim.setNumericType('Temperature')
+              openstudio_schedule.defaultDaySchedule.setName('Heating Day Default')
+              openstudio_schedule.defaultDaySchedule.addValue(OpenStudio::Time.new(0,24,0,0), 100)
+              openstudio_schedule.defaultDaySchedule.setScheduleTypeLimits(openstudio_sch_type_lim)
+              $heating_setpoint_schedule = Honeybee::ScheduleRulesetAbridged.from_schedule_ruleset(openstudio_schedule)
+            end
+            hash[:setpoint][:heating_schedule] = $heating_setpoint_schedule[:identifier]
+          else
+            heating_schedule = thermostat.heatingSetpointTemperatureSchedule.get
+            hash[:setpoint][:heating_schedule] = heating_schedule.nameString
+          end
+          if thermostat.coolingSetpointTemperatureSchedule.empty?
+            # if cooling setpoint schedule is not specified create a new setpoint schedule and assign to HB thermostat object
+            # first check if schedule is already created
+            if $cooling_setpoint_schedule.nil?
+              openstudio_model = OpenStudio::Model::Model.new
+              openstudio_schedule = OpenStudio::Model::ScheduleRuleset.new(openstudio_model)
+              openstudio_schedule.setName('Cooling Schedule Default')
+              openstudio_sch_type_lim = OpenStudio::Model::ScheduleTypeLimits.new(openstudio_model)
+              openstudio_sch_type_lim.setName('Temperature')
+              openstudio_sch_type_lim.setNumericType('Temperature')
+              openstudio_schedule.defaultDaySchedule.setName('Cooling Day Default')
+              openstudio_schedule.defaultDaySchedule.addValue(OpenStudio::Time.new(0,24,0,0), -100)
+              openstudio_schedule.defaultDaySchedule.setScheduleTypeLimits(openstudio_sch_type_lim)
+              $cooling_setpoint_schedule = Honeybee::ScheduleRulesetAbridged.from_schedule_ruleset(openstudio_schedule)
+            end
+            hash[:setpoint][:cooling_schedule] = $cooling_setpoint_schedule[:identifier]
+          else
+            cooling_schedule = thermostat.coolingSetpointTemperatureSchedule.get
+            hash[:setpoint][:cooling_schedule] = cooling_schedule.nameString
+          end
+        end
+        unless thermal_zone.zoneControlHumidistat.empty?
+          humidistat = thermal_zone.zoneControlHumidistat.get
+          unless humidistat.humidifyingRelativeHumiditySetpointSchedule.empty?
+            humidifying_schedule = humidistat.humidifyingRelativeHumiditySetpointSchedule.get
+            hash[:setpoint][:humidifying_schedule] = humidifying_schedule.nameString
+          end
+          unless humidistat.dehumidifyingRelativeHumiditySetpointSchedule.empty?
+            dehumidifying_schedule = humidistat.dehumidifyingRelativeHumiditySetpointSchedule.get
+            hash[:setpoint][:dehumidifying_schedule] = dehumidifying_schedule.nameString
```

Comment:
We need the `sch.to_ScheduleFixedInterval.is_initialized or sch.to_ScheduleRuleset.is_initialized` check here.